<?xml version="1.0" encoding="utf-8"?>
<launch>
  <arg name="need_nodelet_manager" default="true" />
  <arg name="nodelet_manager_name" default="lane_cluster_detection_nodelet" />

  <arg name="input" default="/rs_d435/aligned_depth_to_color/image_raw" />
  <arg name="rgb_image_topic" default="/rs_d435/color/image_rect_color" />
  <arg name="depth_image_topic" default="/rs_d435/aligned_depth_to_color/image_raw" />
  <arg name="camera_info_topic" default="/rs_d435/aligned_depth_to_color/camera_info" />
  <arg name="debug_view" default="true"/>
  <arg name="frame_id" default="target_object"/>
  <arg name="gui" default="true"/>

  <node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager_name)" args="manager" output="screen" if="$(arg need_nodelet_manager)"/>

  <node pkg="nodelet" type="nodelet" name="depth_distance_filter"
        args="load aerial_robot_perception/DepthDistanceFilter $(arg nodelet_manager_name)" output="screen">
    <remap from="~input" to="$(arg input)" />

    <param name="min_depth" value="0.3"/>
    <param name="max_depth" value="1.9"/>
    <param name="use_distance_from_ground" value="true"/>
    <param name="distance_from_ground" value="0.1"/>
  </node>

  <node name="lane_cluster_detection" pkg="nodelet" type="nodelet" args="load aerial_robot_perception/LaneClusterDetection $(arg nodelet_manager_name)">
    <remap from="~input" to="/depth_distance_filter/output"/>
    <remap from="~rgb_img" to="$(arg rgb_image_topic)" />
    <remap from="cam_info" to="$(arg camera_info_topic)" />
    <param name="debug_view" value="$(arg debug_view)"/>
    <param name="frame_id" value="$(arg frame_id)"/>
    <param name="wall_height" value="1.70"/>
  </node>

</launch>
